pkg/eval/builtin_special_test.go

```diff
@@ -147,6 +147,11 @@ func TestTmp(t *testing.T) {
 
 		That("var x; tmp x = y").DoesNotCompile(),
 		That("{ tmp x = y }").DoesNotCompile(),
+
+		That("{ tmp E:X = y; put $E:X }; has-env X").
+			Puts("y", false),
```

Comment:
Also, there doesn't appear to be an equivalent test of `set` so this is a good time to add one. For example:

```
That("{ set E:X = y; put $E:X }; has-env X; put $E:X").
    Puts("y", true, "y"),
```

Reply:
I think `TestSet` is a better place, so I add this case there.